src/main/java/org/apache/commons/jxpath/FunctionLibrary.java

```diff
@@ -77,12 +79,30 @@ public Set getUsedNamespaces() {
      */
     public Function getFunction(String namespace, String name,
             Object[] parameters) {
+        return getFunction(namespace, name, parameters, null);
```

Comment:
I worry that people will continue calling this method on upgrade and continue to be insecure by default. Instead of passing null, I think a better default would be to pass either  the environment-based of a fail-closed (deny all) implementation of JXPathFilter. Same logic applies for other places where we overload the method signatures.

Reply:
Excellent point. Changed all usages to use a default value of the JXPathFilter that is initialized to the filter set by the system property of the allow list. This guarantees consistent behavior among all method calls.